File: sem3/map/Lab/2/src/model/ConstExp.java
package model;

public class ConstExp extends Exp {
    public int value;

    public ConstExp(int value) {
        this.value = value;
    }

    @Override
    public int eval(MyIDictionary<String, Integer> symTable) {
        return value;
    }

    @Override
    public String toString() {
        return String.valueOf(value);
    }
}

File: sem3/map/~craciunf/c-sharp-toy-language-interpreter/models/ConstExp.cs
/* package models; */
using System;
using System.Collections;
using System.Collections.Generic;

namespace ToyLanguageInterpreter {
    public class ConstExp : Exp {
        private int val;

        public ConstExp(int val) {
            this.val = val;
        }

        /* @Override */
        public override int eval(Dictionary <String, int> symTable) {
            return this.val;
        }

        /* @Override */
        public override String toString() {
            return this.val.ToString();
        }
    }
}
